### ISSUE TITLE
compilation errors for primitive type assertions

### ISSUE THREAD

**Issue by scizor666** [hapijs/code#167]
#### Support plan

* *is this issue currently blocking your project?* (yes/no): no
* *is this issue affecting a production system?* (yes/no): yes

#### Context

* *node version*: v12.22.7
* *module version with issue*: 8.0.4
* *last module version without issue*: 8.03
* *environment* (e.g. node, browser, native): node
* *used with* (e.g. hapi application, another framework, standalone, ...):standalone
* *any other relevant information*:

#### What are you trying to achieve or the steps to reproduce?

```ts

function foo(): number | undefined {
 //
}

const value = foo();
expect(value).to.be(123)

function bar(): boolean {
 //
}
expect(bar()).equals(false)
```

#### What was the result you got?

compilation errors like:

```console
Argument of type '123' is not assignable to parameter of type 'never'
Argument of type 'false' is not assignable to parameter of type 'never' 
```

#### What result did you expect?

no compile error

**Comment by kanongil** [comment]
Thanks for the report. This seems to be an issue with the recent #163. I'm looking into a fix.

**Comment by kanongil** [comment]
I made a fix in #168.
